onnxruntime/core/framework/utils.cc

```diff
@@ -62,20 +71,15 @@ static Status CopyMLValue(const DataTransferManager& data_transfer_mgr,
                           const FeedsFetchesManager::MLValueCopyInfo& copy_info,
                           const OrtValue& source_mlvalue,
                           OrtValue& target_mlvalue) {
-  if (copy_info.copy_provider == nullptr) {
-    target_mlvalue = source_mlvalue;
-  } else {
```

Comment:
This is used in the cached scenario by subgraph execution. The new code seems to always copy, so you're unnecessarily copying in that case.